app/src/main/kotlin/openmeteo/WeatherService.kt

```diff
@@ -0,0 +1,51 @@
+package hu.vanio.kotlin.feladat.ms.openmeteo
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import hu.vanio.kotlin.feladat.ms.WeatherAppConfig
+import hu.vanio.kotlin.feladat.ms.exception.ServiceUnavailable
+import org.springframework.stereotype.Service
+import java.io.IOException
+
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+@Service
+class WeatherService(var config: WeatherAppConfig, var objectMapper: ObjectMapper) {
+    private val id = "WEATHER_SERVICE"
+
+    fun getWeatherStatistics(): WeatherForecasts {
+        val request = HttpRequest.newBuilder()
+                .uri(URI(config.appUrl))
+                .GET()
+                .build()
+        val weatherForecast = try {objectMapper.readValue(HttpClient.newHttpClient()
+                .send(request, HttpResponse.BodyHandlers.ofString()).body(), WeatherForecast::class.java)
+        } catch (e: IOException) {
+            throw ServiceUnavailable(id)
+        }
+        return statistics(weatherForecast)
+    }
+
+    companion object {
+        fun statistics(weatherForecast: WeatherForecast): WeatherForecasts {
```

Comment:
Ez a függvény miért lett companion object?

Reply:
Mehetett volna külön service-be is (talán a név sem túl jó, tulajdonképpen ez csak egy mapper).